### ISSUE TITLE
Slackbot clientSigningSecret warning checking wrong config

### ISSUE THREAD

**Issue by dabbeg** [howdyai/botkit#1587]
Running a slackbot and getting this warning: "WARNING: Your bot is operating without recommended security mechanisms in place." though I configured the slackbot like this:
```
const controller = Botkit.slackbot({}).configureSlackApp({
  clientSigningSecret: 'XXX',
});
```

Looking at SlackBot.js I see that this warning message occurs if I do not put clientSigningSecret in the Botkit.slackbot() constructor in line 18. But If I do that the clientSigningSecret will be overwritten with undefined in line 108 resulting in no clientSigningSecret so users will think they have a clientSigningSecret but it is actually undefined.

So in summary if I put clientSigningSecret in Botkit.slackbot() I will have an undefined clientSigningSecret but if I put it in configureSlackApp() I will have a warning message but a correct clientSigningSecret.

**Comment by benbrown** [comment]
don't use `configureSlackApp`, pass the config into the initial function that creates the controller.

**Comment by dabbeg** [comment]
If I put it in the initial function that creates the controller it does not use the clientSigningSecret and the app is unsecure.

Maybe this was unclear in the original description but it doesn't work using the initial controller. ConfigureSlackApp works though but then I get the warning.

**Comment by dabbeg** [comment]
I see so I should move all config into initial controller and remove the configureSlackApp function. Now everything works as expected.

You should probably remove the configureSlackApp from the code and docs as it is only confusing and not helpful.
